relay-metrics/src/protocol.rs

```diff
@@ -419,26 +425,35 @@ fn parse_tags(string: &str) -> Option<BTreeMap<String, String>> {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub struct Metric {
-    /// The name of the metric, i.e. the MRI.
+    /// The metric resource identifier, in short MRI.
     ///
-    /// The metric name historically has been a freeform string. Over time we realized that we can
-    /// use the metric name to encode things like metric unit, type, etc in order to force
-    /// downstream systems to bucket by those properties, without having to update them. This is
-    /// how the concept of the Metrics Resource Identifier was born.
+    /// MRIs follow three core principles:
+
+    /// 1. **Robustness:** Metrics must be addressed via a stable identifier. During ingestion in
+    ///    Relay and Snuba, metrics are preaggregated and bucketed based on this identifier, so it
+    ///    cannot change over time without breaking bucketing.
+    /// 2. **Uniqueness:** The identifier for metrics must be unique across variations of units and
+    ///    metric types, within and across use cases, as well as between projects and
+    ///    organizations.
+    /// 3. **Abstraction:** The user-facing product changes its terminology over time, and splits
+    ///    concepts into smaller parts. The internal metric identifiers must abstract from that,
+    ///    and offer sufficient granularity to allow for such changes.
     ///
-    /// An MRI has the following format: `<type>:<namespace>/<name>@<unit>`
+    /// MRIs have the format `<type>:<ns>/<name>@<unit>`, comprising the following components:
```

Comment:
nit: Why not write`<namespace>` here?

Reply:
we used `ns` in all the other docstrings

Reply:
OK!